### PR TITLE
python-edtf integration

### PR DESCRIPTION
**Pull request by sdreher**
* Using https://github.com/ixc/python-edtf
* Date display format w/tests

### CODE REVIEW

File: footprints/main/models.py [hunk 1]
```diff
@@ -22,11 +24,64 @@
...
+    def fmt_modifier(self, date_obj):
+        if date_obj == 'open':
+            return 'present'
+        if date_obj == 'unknown':
+            return '?'
+
+    def fmt(self, date_obj):
+        result = ''
+
+        if isinstance(date_obj, basestring):
```
Comment (thraxil) [inline footprints/main/models.py]: I would convert this conditional to a guard so the whole else clause can be de-indented:

```
if isinstance(date_obj, basestring):
    return self.fmt_modifier(date_obj)
precision = ...
```

Comment (sdreher) [inline footprints/main/models.py]: Thanks for the observation. I actually went back and forth on this as I was writing it. Is the approach you suggested considered more pythonic?

Comment (thraxil) [inline footprints/main/models.py]: I feel like it's a good practice independent of language. It reduces the nesting of conditionals and makes the code easier to read.

http://refactoring.com/catalog/replaceNestedConditionalWithGuardClauses.html

http://www.informit.com/articles/article.aspx?p=1149121&seqNum=11

Comment (sdreher) [inline footprints/main/models.py]: Makes sense. I think I'm still heeding the "commandment" referenced in the Beck article: "each routine shall have a single entry and a single exit". I'll update the code.

**Pull request merged**